services/ens/api.go

```diff
@@ -0,0 +1,312 @@
+	if err != nil {
+		return nil, nil
+	}
+
+	return contentHash, nil
+}
+
+func (api *API) PublicKeyOf(ctx context.Context, chainID uint64, username string) ([32]byte, [32]byte, error) {
+	empty := [32]byte{}
+	err := validateENSUsername(username)
+	if err != nil {
+		return empty, empty, err
+	}
+
+	resolverAddress, err := api.Resolver(ctx, chainID, username)
+	if err != nil {
+		return empty, empty, err
+	}
+
+	resolver, err := api.contractMaker.newPublicResolver(chainID, resolverAddress)
+	if err != nil {
+		return empty, empty, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	pubKey, err := resolver.Pubkey(callOpts, nameHash(username))
+	if err != nil {
+		return empty, empty, nil
+	}
+
+	return pubKey.X, pubKey.Y, nil
+}
+
+func (api *API) AddressOf(ctx context.Context, chainID uint64, username string) (*common.Address, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return nil, err
+	}
+
+	resolverAddress, err := api.Resolver(ctx, chainID, username)
+	if err != nil {
+		return nil, err
+	}
+
+	resolver, err := api.contractMaker.newPublicResolver(chainID, resolverAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	addr, err := resolver.Addr(callOpts, nameHash(username))
+	if err != nil {
+		return nil, err
+	}
+
+	return &addr, nil
+}
+
+func (api *API) ExpireAt(ctx context.Context, chainID uint64, username string) (*big.Int, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return nil, err
+	}
+
+	registrar, err := api.contractMaker.newUsernameRegistrar(chainID)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	expTime, err := registrar.GetExpirationTime(callOpts, nameHash(username))
+	if err != nil {
+		return nil, err
+	}
+
+	return expTime, nil
+}
+
+func (api *API) Price(ctx context.Context, chainID uint64) (*big.Int, error) {
+	registrar, err := api.contractMaker.newUsernameRegistrar(chainID)
+	if err != nil {
+		return nil, err
+	}
+
+	callOpts := &bind.CallOpts{Context: ctx, Pending: false}
+	price, err := registrar.GetPrice(callOpts)
+	if err != nil {
+		return nil, err
+	}
+
+	return price, nil
+}
+
+func (api *API) Release(ctx context.Context, chainID uint64, from string, gasPrice *big.Int, gasLimit uint64, password string, username string) (string, error) {
+	err := validateENSUsername(username)
+	if err != nil {
+		return "", err
+	}
+
+	registrar, err := api.contractMaker.newUsernameRegistrar(chainID)
+	if err != nil {
+		return "", err
+	}
+
+	txOpts := &bind.TransactOpts{
+		From: common.HexToAddress(from),
+		Signer: func(addr common.Address, tx *types.Transaction) (*types.Transaction, error) {
+			// return types.SignTx(tx, types.NewLondonSigner(chainID), selectedAccount.AccountKey.PrivateKey)
+			return nil, nil
+		},
+		GasPrice: gasPrice,
+		GasLimit: gasLimit,
+	}
+	tx, err := registrar.Release(txOpts, nameHash(username))
```

Comment:
I noticed that when a sendTransaction is executed in status-go, these functions are called:
```
hash, err = api.s.transactor.SendTransaction(sendArgs, verifiedAccount)
if err != nil {
	return
}
go rpcFiltersSrvc.TriggerTransactionSentToUpstreamEvent(hash)
```
The first part seems to be covered by `registrar.Release(txOpts, nameHash(username))`, but it probably makes sense to call the `TriggerTransactionSentToUpstreamEvent` go routine. IIRC it's used so the wallet knows about the transaction and you eventually receive a signal once the transaction is mined

Reply:
Commented for now til I come back to this